LOADING_SUPPORT.md

```diff
@@ -0,0 +1,180 @@
+
+2. Go ahead and either write the top-level `load()` or
+   `load_from_fileobj()` function in your loader - depending on
+   whether your database driver takes a filename/URL or a stream,
+   respectively.
+
+   If your driver takes a filename/URL, follow the redshift driver
+   example on how to translate the URLs passed in `load()` to the
+   right scheme.
+
+   Delegate out the actual generation of the tricky parts of the
+   e.g. SQL statement to another method which takes as input e.g. the
+   `RedshiftUnloadOptions` output of the function you wrote before.
+   This new function should just `raise NotImplementedError` for now.
+
+3. Now we're going to create a first unit test, modeled on
+   `tests/unit/db/mysql/test_mysql_load_options_known#test_load_known_formats`.
+   Verify that it fails due to the plethora of `NotImplementedError`s
+   raised.
+
+4. Start by resolving the `NotImplementedError` raised from your
+   `known_supported_records_formats_for_load()` method.  For now we're
+   going to assume the very best case - that your database will
+   support every single records format we support--or at least the
+   particularly useful ones.  This will no doubt not be true by the time
+   we get done implementing - but the tests will catch that and we'll
+   tweak this down to what's actually supported by the time we're
+   done.  No need to implement anything else yet.
+
+   ```python
+   return [
+       ParquetRecordsFormat(),
+       DelimitedRecordsFormat(variant='bluelabs'),
+       DelimitedRecordsFormat(variant='csv'),
+       DelimitedRecordsFormat(variant='bigquery'),
+       DelimitedRecordsFormat(variant='vertica'),
+   ]
+   ```
+
+5. Run the test and start addressing issues one by one by implementing
+   the function in question.  This will be slow, finicky work - do
+   yourself a favor and leave yourself copious notes as comments
+   pointing back to the original online documentation for the database
+   in question as you go.  Follow the other drivers' use of
+   `cant_handle_hint()` and `quiet_remove()`.  The MySQL loader in
+   particular uses hint typing to ensure we are covering cases
+   correctly, and has written types for the valid input into MySQL's
+   LOAD statement as well.
+
+   To get the test to pass, you may need to tweak the formats slightly
+   (e.g., changing date formats to ones supported or specifying no
+   compression).  You may not be able to support certain variants or
+   formats at all - leave a comment in the method documenting all of
+   these situations and describing why.
+
+6. Write a unit test for `can_load_this_format()` - see
+   `tests/unit/db/mysql/test_mysql_db_driver.py` for an example.  Now
+   implement `can_load_this_format()` in a similar way to other
+   drivers and verify the tests pass.
+
+[done to here]
+7. Figure out which functions are left raising NotImplementedError, if
+   any.  Write a bunch of TODO comments documenting the different
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker